src/action.js

```diff
@@ -1,38 +1,83 @@
+/**
+ * @fileoverview Actions encapsulate the process of resolving an
+ * action creator. Create an action using `Microcosm::push`:
+ */
+
 import Emitter from './emitter'
 import tag from './tag'
-
-import { inherit, isFunction } from './utils'
+import { isFunction } from './utils'
 
 let uid = 0
 
-/**
- * Actions encapsulate the process of resolving an action creator. Create an
- * action using `Microcosm::push`:
- * @constructor
- * @param {Function|string} command
- * @param {string} [status] Starting status
- */
-export default function Action(command, status) {
-  Emitter.call(this)
+class Action extends Emitter {
+  /**
+   * @param {Function|string} command
+   * @param {string} [status] Starting status
+   */
+  constructor(command, status) {
+    super()
+
+    this.id = uid++
+    this.command = tag(command)
+    this.status = 'inactive'
+    this.payload = undefined
+    this.disabled = false
+    this.complete = false
+    this.parent = null
+    this.next = null
+    this.timestamp = Date.now()
+    this.children = []
+
+    if (status) {
+      console.assert(this[status], 'Unexpected task status ' + status)
+      this[status]()
+    }
+  }
+
+  get type() {
```

Comment:
Keep in mind this regarding getters and setters with flow: https://github.com/facebook/flow/issues/2826

Not saying we shouldn't have them, just that I know it's a thing.